src/Controllers/HelloWorldController.cs

```diff
@@ -47,5 +50,76 @@ public IActionResult CreateNewMovie(MovieViewModel viewMovie)
             UnitOfWork.Complete();
             return RedirectToAction("Index");
         }
+
+        public IActionResult Edit(int? id)
+        {
+            if (id == null)
+                return NotFound();
+            var movie = UnitOfWork.MovieRepository.Get((int)id);
+            if (movie == null)
+                return NotFound();
+            ViewData["Title"] = _localizer["EditMovie"];
+            MovieViewModel model = new MovieViewModel();
+            model.Id = movie.Id;
+            model.Genre = movie.Genre;
+            model.Price = movie.Price;
+            model.Title = movie.Title;
+            movie.ReleaseDate = movie.ReleaseDate;
+            return View(model);
+        }
+
+        [HttpPost]
+        public IActionResult Edit(int id, [Bind("Id,Title,ReleaseDate,Genre,Price")] MovieViewModel model)
+        {
+            var movie = UnitOfWork.MovieRepository.Get((int)id);
+            if (movie == null)
+                return NotFound();
+            movie.Genre = model.Genre;
+            movie.ReleaseDate = model.ReleaseDate;
+            movie.Price = model.Price;
+            movie.Title = model.Title;
+            UnitOfWork.MovieRepository.Update(movie);
+            UnitOfWork.Complete();
+            return RedirectToAction("Index");
+        }
+
+        public IActionResult Delete(int? id)
+        {
+            if (id == null)
+                return NotFound();
+            var movie = UnitOfWork.MovieRepository.Get((int)id);
+            if (movie == null)
+                return NotFound();
+            MovieViewModel model = new MovieViewModel();
+            model.Id = (int) id;
+            return View(model);
+        }
+
+        [HttpPost, ActionName("Delete")]
+        public IActionResult DeleteConfirmed(int id)
+        {
+            var movie = UnitOfWork.MovieRepository.Get((int)id);
+            if (movie == null)
+                return NotFound();
+            UnitOfWork.MovieRepository.Remove(movie);
+            UnitOfWork.Complete();
+            return RedirectToAction("Index");
+        }
+
+        public IActionResult Details(int? id)
```

Comment:
this method is like Edit method, it will be great abstract the similar code to a helper method